### DISCUSSION TITLE
Automatic resolution of repeat regions in bacterial genome hybrid assembly (isolate)

### DISCUSSION THREAD

**Discussion by felince** [ablab/spades#1337] [Q&A]
Hi,

I'm trying to assemble a bacterial genome including its plasmid using the isolate option with long (PacBio) and short (DNBSEQ) reads. Before we had attempted to sequence using only the PacBio reads and we had trouble assembling the plasmid, but this was solved using the hybrid approach with Spades. However, it seems that even with the long read data to solve the repeat regions there are still many loops in the assembly that I guess must be corrected manually. The plasmid itself had a repeat region that I corrected manually, but for the chromosome this will be a never-ending task. Here's a picture of how the assembly looks at the moment. The green nodes are the plasmid. The blue nodes appeared as a separate contig in the long-read data only assembly and now appear correctly (I think) integrated within the chromosome.

![image](https://github.com/user-attachments/assets/a59712a2-50fa-4d5e-b98c-2fd808adaec5)

By the way, using only the long reads, the chromosome was assembled perfectly circular (albeit this was done by a third party using HiFiasm). Am I missing something in the assembly parameters to solve these regions automatically? The coverage of the long-read data is relatively low (about 30x) so maybe this is a reason why this data is not being used to solve the repeat regions. Any advice is welcome. Thank you.

**Comment by asl** [comment]
Looks like there is some strain heterogeneity here, no? Are these long stretches of bubbles reflect the real strain differences?

**Reply by felince** [reply]
There is some heterogeneity in the population as sometimes the plasmid is lost in some of the cells. However, the cells used for sequencing come from the same glycerol stock of a pure colony that was previously isolated. The culture batches used for long and short read sequencing experiments are different as we noticed that the long read data was not enough, and we had to obtain more DNA for the short-read sequencing which was done at a later time. Thank you.

**Reply by asl** [reply]
Is it possible for you to attach graph here?

**Reply by felince** [reply]
Unfortunately, at the moment I can't upload the graph :(. Is there something that I should look for in the graph that could help me solve the bubbles?

**Reply by asl** [reply]
Well, the first thing is that you need to understand their origin. As usually they represent some variation. And if this is a real variation, it might be possible that long reads do not span it (as they might have different variants in these regions as well). But this is just wild guess from this small picture.

**Reply by felince** [reply]
Yes, I think there is some real variation in the population. Also, in other sequencing experiments we have observed variation in the length of some tandem repeats for other strains. However, I would assume that there's a variant that is contained within the majority of the cells and this would be used to solve the majority of the bubbles. As I mentioned the repeat that was not solved in the plasmid was relatively easy for me to solve manually just by mapping the reads back to the nodes, so I thought that the adjustment of one parameter in the assembly command could help me deal with the majority of the cases. It is my first time using Spades for hybrid data so that's why I wondered if my assembly parameters were not the best. By the way, I used the isolate option as suggested by spades after using the plasmid option or no additional options.

**Reply by asl** [reply]
Well, if both of variants are sufficiently abundant there will be no "variant collapsing". After all, it is not possible in general to distinguish such variants from diverged repeats, for example. Isolate / multi-cell mode expect some heterogeneity here.

You might want to give metagenomic mode a try via `--meta --only-assembler`. It might collapse these variants basing on IDY and other heuristics (in general this is not safe, but in most cases fine for metagenome assemblies as they are consensus across strains anyway).

**Reply by felince** [reply]
Ok, I will try to use the metagenomic mode. The repeats that I am observing are in most cases 300 bp of length or less. In few cases they reach around 1000 bp, but they are mostly short. The machine I use for assembly is different from the one I'm using right now so let me try to assemble and I'll get back with the results later. Thank you for your help.
